Sources/RedisKit/RedisSet.swift

```diff
@@ -1,143 +1,173 @@
+    ///     let setOfIDs = client.makeSetReference(key: String, type: Int.self)
+    ///     // setOfIDs represents a Set of `Int`.
     ///
-    /// - Parameter fromKey: The key to identify the Set in this `RedisClient`.
-    /// - Parameter ofType: The type of the elements contained within the set.
-//    public func createSetReference<T>(fromKey key: String, ofType type: T.Type) -> RedisSet<T> {
-//        return RedisSet(identifier: key, using: self)
-//    }
+    /// - Parameters:
+    ///     - key: The Redis key to identify the set.
+    ///     - type: The Swift type representation of the elements in the set.
+    /// - Returns: A `RedisSet` for the key and element type specified.
+    @inlinable
+    public func makeSetReference<T>(key: String, type: T.Type = T.self) -> RedisSet<T> {
+        return RedisSet(identifier: key, client: self)
+    }
 
-    /// Creates a `RedisSet` for the provided key, representing a `Collection` type.
+    /// Creates a `RedisSet` reference to the value stored at `key` with values of `type`.
     ///
-    ///     let idSet = redis.createSetReference(fromKey: "ids", ofType: [Int].self)
-    ///     // idSet represents a Set of `Int`.
+    ///     let setOfIDs = client.makeSetReference(key: String, type: [Int].self)
+    ///     // setOfIDs represents a Set of `Int`.
     ///
-    /// - Parameter fromKey: The key to identify the Set in this `RedisClient`.
-    /// - Parameter ofType: The `Collection` type this set represents.
-//    public func createSetReference<C: Collection>(fromKey key: String, ofType type: C.Type) -> RedisSet<C.Element>
-//        where C.Element: RESPValueConvertible
-//    {
-//        return RedisSet(identifier: key, using: self)
-//    }
-//}
+    /// - Parameters:
+    ///     - key: The Redis key to identify the set.
+    ///     - type: The Swift type representation of the elements in the set.
+    /// - Returns: A `RedisSet` for the key and element type specified.
+    @inlinable
+    public func makeSetReference<C: Collection>(
+        key: String,
+        type: C.Type = C.self
+    ) -> RedisSet<C.Element> {
+        return RedisSet(identifier: key, client: self)
+    }
+}
 
 /// A reference to a specific Set in a Redis instance.
 ///
 /// https://redis.io/topics/data-types-intro#sets
-//public struct RedisSet<Element> where Element: RESPValueConvertible {
-//    private let id: String
-//    private let redis: RedisDatabase
-
-    /// - Parameter identifier: The key identifier to reference this set.
-    /// - Parameter using: The connection pool to use for interacting with this set reference.
-//    public init(
-//        identifier: String,
-//        using redis: RedisDatabase
-//    ) {
-//        self.id = identifier
-//        self.redis = redis
-//    }
-
-    /// Returns the total count of elements in the set.
+public struct RedisSet<Element> where Element: RESPValueConvertible {
+    @usableFromInline let id: String
+    @usableFromInline let client: RedisClient
```

Comment:
I think having attributes on a newline before is a bit more readable. But I'm not sure which style is more Swifty.

Reply:
I don't think either is "Swifty" and it's definitely a preference :P

I can update this so it's consistent

Reply:
This one is officially a `LongAttribute`, so would automatically be placed on its own line when generated.

https://github.com/apple/swift/blob/master/include/swift/AST/Attr.def#L321